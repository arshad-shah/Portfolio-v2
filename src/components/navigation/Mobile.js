import React from "react";
import { IconButton } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { Link } from "react-scroll";
import Resume from "../../assets/ArshadResume.pdf";
import { Divide as Hamburger } from "hamburger-react";
import { slide as Menu } from "react-burger-menu";
import { Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
  button: {
    "&:hover": {
      backgroundColor: "transparent",
    },
    "&:active": {
      backgroundColor: "transparent",
    },
    "&:visited": {
      backgroundColor: "transparent",
    },
    "&:link": {
      backgroundColor: "none",
    },
    color: "#000000",
    padding: "0.7rem",
    textDecoration: "none",
    cursor: "pointer",
    borderRadius: "0.5rem",
    transition: "all 0.5s ease",
  },
}));

var styles = {
  bmMenuWrap: {
    position: "fixed",
    height: "35%",
    top: "4.5rem",
  },
  bmMenu: {
    background: "#fdff95",
    padding: "0.5rem",
    borderRadius: "0 0 0 1rem",
    border: "1px solid black",
  },
  bmMorphShape: {
    fill: "#fdff95",
  },
  bmItemList: {
    color: "#fdff95",
    display: "flex",
    flexDirection: "column",
    alignContent: "center",
    alignItems: "stretch",
    justifyContent: "space-around",
  },
  bmItem: {
    display: "inherit",
  },
  bmOverlay: {
    background: "transparent",
  },
};

export default function Mobile({ click, closeMobileMenu, handleMenu }) {
  const classes = useStyles();

  var prevScrollpos = window.pageYOffset;
  window.onscroll = function () {
    var currentScrollPos = window.pageYOffset;
    if (prevScrollpos < currentScrollPos) {
      closeMobileMenu();
    }
    prevScrollpos = currentScrollPos;
  };

  return (
    <>
      <IconButton
        role="button"
        color="default"
        className={classes.menuButton}
        aria-label="menu"
        onClick={handleMenu}
        data-block="hamburgermenubuttonmobile"
      >
        <Hamburger
          distance="md"
          toggled={click}
          size={40}
          rounded
          color="#c78f00"
        />
      </IconButton>

      <Menu
        right
        isOpen={click}
        customBurgerIcon={false}
        customCrossIcon={false}
        width={180}
        styles={styles}
      >
        <Link
          className={classes.button}
          data-block="Aboutmemenuitemmobile"
          to="aboutme"
          spy={true}
          smooth={true}
          onClick={() => closeMobileMenu()}
        >
          <Typography variant="button">About Me</Typography>
        </Link>
        <Link
          className={classes.button}
          data-block="Projectsmenuitemmobile"
          to="projects"
          spy={true}
          smooth={true}
          onClick={() => closeMobileMenu()}
        >
          <Typography variant="button">Projects</Typography>
        </Link>
        <a
          className={classes.button}
          onClick={() => closeMobileMenu()}
          variant="outlined"
          href={Resume}
          data-block="Resumemenuitemmobile"
        >
          <Typography variant="button">Resume</Typography>
        </a>
      </Menu>
    </>
  );
}
