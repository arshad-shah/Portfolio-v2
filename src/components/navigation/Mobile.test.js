import React from "react";
import { render, screen } from "@testing-library/react";
import Mobile from "./Mobile";

describe("Mobile of the site:", () => {
  it("should render Mobile menu with the opening hamburger button", () => {
    render(<Mobile />);
    const menuIcon = screen.getByRole("button");
    expect(menuIcon).toBeInTheDocument();
  });

  it("should render Mobile", () => {
    const { container } = render(<Mobile />);
    expect(container).toMatchSnapshot();
  });
});
