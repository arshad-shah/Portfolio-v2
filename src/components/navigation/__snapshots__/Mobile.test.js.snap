// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile of the site: should render Mobile 1`] = `
<div>
  <button
    aria-label="menu"
    class="MuiButtonBase-root MuiIconButton-root"
    data-block="hamburgermenubuttonmobile"
    role="button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <div
        class="hamburger-react"
        role="button"
        style="cursor: pointer; height: 48px; position: relative; transition: 0.4s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none;"
        tabindex="0"
      >
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 4px; position: absolute; border-radius: 9em 0 0 9em; width: 20px; top: 11px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 50%; position: absolute; border-radius: 0 9em 9em 0; width: 20px; top: 11px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 4px; position: absolute; border-radius: 9em 0 0 9em; width: 20px; top: 23px; transition: 0.4s cubic-bezier(0, 0, 0, 1); opacity: 1; transform: none;"
        />
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 50%; position: absolute; border-radius: 0 9em 9em 0; width: 20px; top: 23px; transition: 0.4s cubic-bezier(0, 0, 0, 1); opacity: 1; transform: none;"
        />
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 4px; position: absolute; border-radius: 9em 0 0 9em; width: 20px; top: 35px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
        <div
          style="background: rgb(199, 143, 0); height: 3px; left: 50%; position: absolute; border-radius: 0 9em 9em 0; width: 20px; top: 35px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
        />
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div>
    <div
      class="bm-overlay"
      style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: transparent; opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
    />
    <div
      aria-hidden="true"
      class="bm-menu-wrap"
      id=""
      style="position: fixed; right: 0px; z-index: 1100; width: 180px; height: 35%; transform: translate3d(100%, 0, 0); transition: all 0.5s; top: 4.5rem;"
    >
      <div
        class="bm-menu"
        style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(253, 255, 149); padding: 0.5rem; border-radius: 0 0 0 1rem; border: 1px solid black;"
      >
        <nav
          class="bm-item-list"
          style="height: 100%; color: rgb(253, 255, 149); display: flex; flex-direction: column; align-content: center; align-items: stretch; justify-content: space-around;"
        >
          <a
            class="bm-item makeStyles-button-6"
            data-block="Aboutmemenuitemmobile"
            style="display: inherit;"
            tabindex="-1"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              About Me
            </span>
          </a>
          <a
            class="bm-item makeStyles-button-6"
            data-block="Projectsmenuitemmobile"
            style="display: inherit;"
            tabindex="-1"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              Projects
            </span>
          </a>
          <a
            class="bm-item makeStyles-button-6"
            data-block="Resumemenuitemmobile"
            href="ArshadResume.pdf"
            style="display: inherit;"
            tabindex="-1"
            variant="outlined"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              Resume
            </span>
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
